src/meta/src/hummock/manager/context.rs

```diff
@@ -310,6 +318,33 @@ fn check_sst_retention(
     Ok(())
 }
 
+async fn check_gc_history(
+    db: &DatabaseConnection,
+    // need IntoIterator to work around stream's "implementation of `std::iter::Iterator` is not general enough" error.
+    object_ids: impl IntoIterator<Item = HummockSstableObjectId>,
+) -> Result<()> {
+    let futures = object_ids.into_iter().map(|id| async move {
+        let id: risingwave_meta_model::HummockSstableObjectId = id.try_into().unwrap();
+        hummock_gc_history::Entity::find_by_id(id)
+            .one(db)
+            .await
+            .map_err(Error::from)
+    });
+    let res: Vec<_> = stream::iter(futures).buffer_unordered(10).collect().await;
+    let res: Result<Vec<_>> = res.into_iter().collect();
+    let expired_object_ids: Vec<_> = res?.into_iter().flatten().collect();
```

Comment:
Only needs to determine if the element exists, not all the results, can we replace collect with next ?